tests/unit_tests/test_protocols.py

```diff
@@ -0,0 +1,135 @@
+"""Unit tests for the protocols module."""
+
+from pathlib import Path
+
+from emodelrunner.load import (
+    load_config,
+    get_prot_args,
+)
+from emodelrunner.create_cells import create_cell_using_config
+from emodelrunner.protocols.create_protocols import ProtocolBuilder
+from emodelrunner.protocols.reader import ProtocolParser
+from emodelrunner.synapses.create_locations import get_syn_locs
+
+from tests.utils import cwd
+
+sscx_sample_dir = Path("examples") / "sscx_sample_dir"
+thalamus_sample_dir = Path("examples") / "thalamus_sample_dir"
+
+# fmt: off
```

Comment:
here again: what is fmt?

Reply:
It's to protect these lines from black's formatting

```
sscx_recipe_protocol_keys = {
    "bAP", "APWaveform_360", "RMP", "SpikeRec_all", "ThresholdDetection", "Step_200",
    "Step_140", "Rin", "IV_-100", "Main", "RinHoldcurrent", "Step_280"}

thalamus_recipe_protocol_keys = {
    "ThresholdDetection_hyp", "Rin_dep", "ThresholdDetection_dep", "Rin_hyp", "Main",
    "RinHoldcurrent_hyp", "RMP", "Step_150", "Step_200", "Step_200_hyp", "RinHoldcurrent_dep"}
```